src/shmem_node_util.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+
+#include "shmem_internal.h"
+#include "shmem_node_util.h"
+
+
+int *shmem_internal_location_array = NULL;
+
+static int node_util_is_initialized = 0;
+static char my_hostname[SHMEM_INTERNAL_MAX_HOSTNAME_LEN];
+static int n_local_pes = 0;
+
+int shmem_node_util_init(void)
+{
+    int ret;
+    char errmsg[256];
+
+    if (!node_util_is_initialized) {
```

Comment:
I allow init to be called multiple times, for example, because `STX_AUTO` and `ON_NODE_COMMS` both require init, but init only needs to be called once.  Alternatively, I could  first check `if (!shmem_node_util_is_initialized())` and throw an error if init is called more than once like you describe.  Is that what you prefer?

I think startup should probably have a similar behavior, right?  I must have missed that.